motionplan/constraint.go

```diff
@@ -504,3 +427,61 @@ func NewOctreeCollisionConstraint(octree *pointcloud.BasicOctree, threshold int,
 	}
 	return constraint
 }
+
+// Given a constraint input with only frames and input positions, calculates the corresponding poses as needed.
+func resolveStatesToPositions(state *ik.State) error {
+	if state.Position == nil {
+		if state.Frame != nil {
+			if state.Configuration != nil {
+				pos, err := state.Frame.Transform(state.Configuration)
+				if err == nil {
+					state.Position = pos
+				} else {
+					return err
+				}
+			} else {
+				return errors.New("invalid constraint input")
+			}
+		} else {
+			return errors.New("invalid constraint input")
+		}
+	}
+	return nil
+}
+
+// Given a constraint input with only frames and input positions, calculates the corresponding poses as needed.
+func resolveSegmentsToPositions(segment *ik.Segment) error {
+	if segment.StartPosition == nil {
+		if segment.Frame != nil {
+			if segment.StartConfiguration != nil {
+				pos, err := segment.Frame.Transform(segment.StartConfiguration)
```

Comment:
[nit] could be simplified into 
```
if pos, err := segment.Frame.Transform(segment.StartConfiguration); err != nil {
    return err
} else {
    segment.StartPosition = pos
}
```

Reply:
That panics if `StartConfiguration` is nil, which it is allowed to be

Reply:
i don't think it would panic since this would be below the `if segment.StartConfiguration != nil {` line

Reply:
No, you're right, sorry; I misinterpreted what you were saying.